bundles/org.openhab.binding.hdpowerview/src/main/java/org/openhab/binding/hdpowerview/internal/HDPowerViewWebTargets.java

```diff
@@ -64,128 +59,160 @@
+        String json = invoke(HttpMethod.GET, scenes, null, null);
         return gson.fromJson(json, Scenes.class);
     }
 
     /**
      * Instructs the hub to execute a specific scene
-     * 
+     *
      * @param sceneId id of the scene to be executed
      * @throws ProcessingException if there is any processing error
      * @throws HubMaintenanceException if the hub is down for maintenance
      */
     public void activateScene(int sceneId) throws ProcessingException, HubMaintenanceException {
-        WebTarget target = sceneActivate.queryParam(SCENE_ID, sceneId);
-        invoke(target.request().header(CONN_HDR, CONN_VAL).buildGet(), target, null);
+        invoke(HttpMethod.GET, sceneActivate, Query.of("sceneId", Integer.toString(sceneId)), null);
     }
 
-    private synchronized String invoke(Invocation invocation, WebTarget target, @Nullable String jsonCommand)
-            throws ProcessingException, HubMaintenanceException {
+    /**
+     * Invoke a call on the hub server to retrieve information or send a command
+     *
+     * @param method GET or PUT
+     * @param url the host url to be called
+     * @param query the http query parameter
+     * @param jsonCommand the request command content (as a json string)
+     * @return the response content (as a json string)
+     * @throws ProcessingException
+     * @throws HubMaintenanceException
+     */
+    private synchronized String invoke(HttpMethod method, String url, @Nullable Query query,
+            @Nullable String jsonCommand) throws ProcessingException, HubMaintenanceException {
         if (logger.isTraceEnabled()) {
-            logger.trace("API command {} {}", jsonCommand == null ? GET : PUT, target.getUri());
+            logger.trace("API command {} {}", method, url);
             if (jsonCommand != null) {
                 logger.trace("JSON command = {}", jsonCommand);
             }
         }
-        Response response;
+        Request request = httpClient.newRequest(url).method(method).header("Connection", "close").accept("*/*");
+        if (query != null) {
+            request.param(query.key(), query.value());
+        }
+        if (jsonCommand != null) {
+            request.header(HttpHeader.CONTENT_TYPE, "application/json").content(new StringContentProvider(jsonCommand));
+        }
+        ContentResponse response;
         try {
-            response = invocation.invoke();
-        } catch (ProcessingException e) {
+            response = request.send();
+        } catch (InterruptedException | TimeoutException | ExecutionException e) {
             if (Instant.now().isBefore(maintenanceScheduledEnd)) {
                 // throw "softer" exception during maintenance window
                 logger.debug("Hub still undergoing maintenance");
                 throw new HubMaintenanceException("Hub still undergoing maintenance");
             }
-            throw e;
+            throw new ProcessingException(e.getMessage());
```

Comment:
Can you include the cause exception here? Although another question is why you are still using the jaxrs exception when you aren't using the jaxrs framework.

Reply:
I just wanted to minimise the changes on the calling class.
But I will change it.